website2/src/components/documentationPage/documentHeader/index.tsx

```diff
@@ -1,30 +1,31 @@
 import { Text } from "@/components/text";
-import { PropsWithChildren, ReactNode } from "react";
+import { ReactNode } from "react";
 import styles from "./styles.module.css";
 import { Share } from "@/components/icons/share";
 import Link from "next/link";
+import { z } from "zod";
+import { withPropCheck } from "@/utils/withPropCheck";
 
-type Props = PropsWithChildren & {
-  type: "h1" | "h2" | "h3";
-  className?: string;
-};
+const PropSchema = z.object({
+  type: z.enum(["h1", "h2", "h3"]),
+  className: z.string().optional(),
+  children: z.custom(),
+});
 
-export const DocumentHeader = ({ type, children, ...rest }: Props) => {
-  const id = getId(children);
-  return (
-    <Text
-      id={id}
-      type={type}
-      {...rest}
-      className={`${rest.className} ${styles.container}`}
-    >
-      <Link href={`#${id}`} className={styles.link}>
-        <span>{children}</span>
-        <Share className={styles.share} />
-      </Link>
-    </Text>
-  );
-};
+export const DocumentHeader = withPropCheck<z.infer<typeof PropSchema>>(
```

Comment:
Can `T` be inferred here from the first argument? If not I think maybe going the other way of making `T` the type of `PropSchema` and using `z.infer` in the signature of the passed in component should let typescript auto-infer this so we don't have to write the generic all the time

Reply:
yea it can be, i'll remove it from places